src/coreclr/jit/emitarm64.cpp

```diff
@@ -8473,7 +8470,7 @@ void emitter::emitIns_J(instruction ins, BasicBlock* dst, int instrCount)
 
     id->idIns(ins);
     id->idInsFmt(fmt);
-    id->idjShort = idjShort;
+    id->idjShort = false;
```

Comment:
When branching to a `finally` funclet on ARM64, we emit an `INS_bl_local` instruction -- previously, we could guarantee this jump would stay within the hot section, and thus would set `idjShort = true`. Now that EH funclets can be moved to the cold section, we can no longer assume this, and must set `idjKeepLong` based on if we're crossing hot/cold sections. Without changing this logic, we'll hit asserts in `emitOutputLJ` due to the branch being too short.